molr-mole-core/src/test/java/io/molr/mole/core/tree/DisposeMissionsTests.java

```diff
@@ -0,0 +1,136 @@
+package io.molr.mole.core.tree;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import io.molr.commons.domain.AgencyState;
+import io.molr.commons.domain.Mission;
+import io.molr.commons.domain.MissionCommand;
+import io.molr.commons.domain.MissionHandle;
+import io.molr.commons.domain.MissionInstance;
+import io.molr.commons.domain.MissionState;
+import io.molr.commons.domain.StrandCommand;
+import io.molr.mole.core.api.Mole;
+import io.molr.mole.core.runnable.RunnableLeafsMission;
+import io.molr.mole.core.runnable.RunnableLeafsMole;
+import io.molr.mole.core.runnable.lang.Branch;
+import io.molr.mole.core.runnable.lang.RunnableLeafsMissionSupport;
+
+public class DisposeMissionsTests {
+
+    private final static Logger LOGGER = LoggerFactory.getLogger(DisposeMissionsTests.class);
+    
+    Mole mole;
+    Mission demoMission = new Mission(demoMission().name());
+
+    @Before
+    public void prepareMoleAndMissions() {
+        Set<RunnableLeafsMission> missions = Sets.newHashSet();
+        missions.add(demoMission());
+        mole = new RunnableLeafsMole(missions);
+    }
+
+    @Test
+    public void ckeckThatDisposeCommandIsAllowedIffMissionIsCompleted() {
+        MissionHandle instance1Handle = mole.instantiate(demoMission, Maps.newHashMap()).block();
+        MissionHandle instance2Handle = mole.instantiate(demoMission, Maps.newHashMap()).block();
+
+        sleepUnchecked(1000);
+
+        MissionState instance1StateAfterInstantiating = mole.statesFor(instance1Handle).blockFirst();
+        assertFalse(instance1StateAfterInstantiating.allowedMissionCommands().contains(MissionCommand.DISPOSE));
+
+        MissionState instance2StateAfterInstantiating = mole.statesFor(instance2Handle).blockFirst();
+        assertFalse(instance2StateAfterInstantiating.allowedMissionCommands().contains(MissionCommand.DISPOSE));
+
+        mole.instructRoot(instance1Handle, StrandCommand.RESUME);
+        sleepUnchecked(1000);
+
+        MissionState instance1StateAfterMissionCompleted = mole.statesFor(instance1Handle).blockFirst();
+        assertTrue(instance1StateAfterMissionCompleted.allowedMissionCommands().contains(MissionCommand.DISPOSE));
```

Comment:
assertTrue() is always a bit sparse on information in the failure case...

consider using assertj assertions:
```
Assertions.assertThat(instance1StateAfterMissionCompleted.allowedMissionCommands()).containsExactly(DISPOSE);
```
... etc... this will give you nice messages in the failure cases.